devito/ir/equations/equation.py

```diff
@@ -147,8 +148,11 @@ def __new__(cls, *args, **kwargs):
                  for k, v in mapper.items() if k}
         dspace = DataSpace(dintervals, parts)
 
+        # Lower all Differentiable operations into SymPy operations
+        rhs = diff2sympy(expr.rhs)
```

Comment:
note for reviewers: avoids minor redundant expressions in the generated code, such as `a + b` and `a + b` where the former was of type Differentiable and the latter of type sympy. And even though Differentiable inherits from sympy, it would still be the case that `a + b != a + b` . This was BTW causing a minor op-count discrepancy in one of the tests

Reply:
Not done on lhs?

Reply:
at this point the lhs is already a `types/...` so it's fine